### PR TITLE
- adds an api key authentication provider

### PR DESCRIPTION
**Pull request by baywet**
related https://github.com/microsoft/kiota/issues/1902

### CODE REVIEW

File: packages/abstractions/test/common/authentication/apiKeyAutenticationProvider.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      "key",
+      "param",
+      ApiKeyLocation.Header
+    );
+    const request = new RequestInformation();
+    request.urlTemplate = "https://localhost{?param1}";
+    await provider.authenticateRequest(request);
+    assert.equal(request.URL, "https://localhost");
+    assert.equal(request.headers["param"], "key");
+  });
```
Comment (gavinbarron) [inline packages/abstractions/test/common/authentication/apiKeyAutenticationProvider.ts:72]: Can we add a test that validates the error throwing behavior of the authenticate request?

**Review by gavinbarron — DISMISSED** [review]
This looks really good! Adding a test for the error throwing behavior would improve our test coverage.

File: packages/abstractions/test/common/authentication/apiKeyAutenticationProvider.ts (hunk 1 above)
Comment (baywet) [reply packages/abstractions/test/common/authentication/apiKeyAutenticationProvider.ts:80]: done, also moved the defensive programing earlier in the flow.

**Review by nikithauc — APPROVED** [review]

**Comment by nikithauc** [comment]
Oops! I did not observe if auto-merge was set on this PR.
I did not intend to override @gavinbarron's comment.

**Comment by baywet** [comment]
no worries, I had time to address the comment before you left your review.

**Pull request merged**